### PR TITLE
[PORT] Smooth Doors from Yogstation

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

This PR ports https://github.com/yogstation13/Yogstation/pull/8901 from monster860 it adds a smoother door animation to every airlock and i guess you could say that i am partially porting https://github.com/yogstation13/Yogstation/pull/9705 trough i only add a cut_overlays().

https://streamable.com/927o6y (outdated)Some airlocks i put there that show some of the new airlock movements
https://streamable.com/w3pum7 a bit speeded up

## Why It's Good For The Game

Smoother Door animation

## Changelog
:cl: 
tweak: Doors are smoother
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
This kind of looks cursed not gonna lie, good PR.

**Comment by ike709** [comment]
I feel like it's _slightly_ too slow, but otherwise it looks good.

**Comment by MNarath1** [comment]
> I feel like it's _slightly_ too slow, but otherwise it looks good.

i will see what i can do about the speed i think i got an idea how to fix this and make it a bit  faster

**Comment by MNarath1** [comment]
not sure if i really have to link the PR for a such minor change like adding a cut_overlays() but i am letting it stay for now

**Comment by PowerfulBacon** [comment]
this looks absolutely cursed, does this get shutters like in cargo too?

**Comment by PowerfulBacon** [comment]
holy shit the alien door is sexy

**Comment by MNarath1** [comment]
> holy shit the alien door is sexy

is the abductor airlock even used anywhere in the game?

**Comment by MNarath1** [comment]
ok scrap that it is on a single emergency shuttle i think

**Comment by MNarath1** [comment]
> this looks absolutely cursed, does this get shutters like in cargo too?

I give in the slower version looked pretty cursed but i think after the speed up it looks pretty good

**Comment by MNarath1** [comment]
Also tell me if you think its too fast now or if its fine

**Comment by Chessus** [comment]
Looks lovely!

**Comment by ike709** [comment]
Looks good.

**Comment by Chessus** [comment]
I'm going to sound like a noob but are door opening  animation purely cosmetic? Or when the animation is playing the code says no player can pass?

**Comment by MNarath1** [comment]
> I'm going to sound like a noob but are door opening animation purely cosmetic? Or when the animation is playing the code says no player can pass?

from what i can tell the moment you can pass is determinated when the density var is set to False this seems to be handled by some sleep() that delay the activation of the next part

**Comment by MNarath1** [comment]
actually should i change the sleep() var so that it matches up more with the new animation time or not? at the moment it would be around 1.5 deciseconds off

**Comment by MNarath1** [comment]
the difference is very small but still kinda noticable i think might just imagine it its still just 1.5 deciseconds so this would buff how fast you can go trough doors but only by like 0.15 seconds so yeah

**Comment by MNarath1** [comment]
yeah i think its better that way seems like it was also setup to match the old animation speed before so doing that might be good

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -487,144 +528,116 @@
...
-	var/mutable_appearance/frame_overlay
-	var/mutable_appearance/filling_overlay
-	var/mutable_appearance/lights_overlay
-	var/mutable_appearance/panel_overlay
-	var/mutable_appearance/weld_overlay
-	var/mutable_appearance/damag_overlay
-	var/mutable_appearance/sparks_overlay
-	var/mutable_appearance/note_overlay
-	var/notetype = note_type()
+	for(var/obj/effect/overlay/airlock_part/part in part_overlays)
```
Comment (ike709) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
	for(var/obj/effect/overlay/airlock_part/part as() in part_overlays)
```

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -1165,7 +1182,7 @@
 	sleep(1)
 	set_opacity(0)
 	update_freelook_sight()
-	sleep(4)
+	sleep(2.5)
```
Comment (ike709) [inline code/game/machinery/doors/airlock.dm]: You also need to change the sleep in `close()`. Might be a good idea to replace the values with a var on the airlock so people don't forget about this if we tweak it in the future.

File: code/game/objects/effects/overlays.dm [hunk 3]
```diff
@@ -52,6 +52,23 @@
...
+	layer = FLOAT_LAYER - 1
+	vis_flags = VIS_INHERIT_ID
+	var/side_id
+	var/open_px = 0
+	var/open_py = 0
+	var/move_start_time = 0 // for opening; closing uses reversed.
+	var/move_end_time = 3.5
+	var/aperture_angle = 0
+	var/obj/machinery/door/airlock/parent
+// in case some caveman is still using 512
```
Comment (ike709) [inline code/game/objects/effects/overlays.dm]: You can't compile bee on 512 nor play bee on 512, can this just be removed?

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (MNarath1) [reply code/game/machinery/doors/airlock.dm]: implemented

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (MNarath1) [reply code/game/machinery/doors/airlock.dm]: var for both sleep in open() and close() also implemented

File: code/game/objects/effects/overlays.dm (hunk 3 above)
Comment (MNarath1) [reply code/game/objects/effects/overlays.dm]: Removed the unneeded code for 512 support

**Comment by Chessus** [comment]
Just tested it and looks great. You zoom through fire doors also

**Comment by MNarath1** [comment]
> Just tested it and looks great. You zoom through fire doors also

well the firelocks are not influence by this PR the firelocks are as the airlocks subtypes of doors but they are not subtypes of airlocks so changes at airlocks should not influence them so this is not done by my this PR

**Comment by MNarath1** [comment]
Just to make sure i tested it on master and firelocks are exactly as fast

**Comment by Chessus** [comment]
Ah I see, must just be becuase im on a local sever and there's no laggg

**Pull request merged**